### PR TITLE
BE | Blog Listing

### PR DESCRIPTION
**Pull request by engineervix**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1347772598)

### Description of Changes Made

This is the backend counterpart to #40 and primarily consists of the following changes:

- removal of the `intro` field from `BlogIndexPage`
- removeal of the `related_links` reverse relation for `BlogIndexPage`
- removal of the `get_popular_tags` method on `BlogIndexPage`
- updating the `blog_posts` context variable by adding extra keys to each item
- refactoring of the pagination mechanism (we're no longer using AJAX) and also account for query params in pagination
- cleanup of the `BlogPage` model by removing unused reverse relation: `blog_page_tag_select` (this ideally should have been done as part of #35) 

In addition, the pagination template has been updated with the logic required to replicate the design:

image

### How to Test

Got to <http://localhost:8000/blog/>. 

### Screenshots

<details>
  <summary>Expand to see more</summary>

https://github.com/torchbox/torchbox.com/assets/7713776/dd65a2e5-aeb9-4c78-a7a8-01a12fd3674e

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on Linux (Ubuntu)
  - Latest version of Firefox on Linux (Ubuntu)

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] New font variants have not been added
- [x] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/project_styleguide/templates/patterns/molecules/pagination/pagination.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 context:
-  paginator_page:
+  pages:
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/molecules/pagination/pagination.yaml:2]: I think the yaml isn't quite correct here as it isn't showing the current page (and we need to add it to the blog listing page too) but I can fix that in the front-end ticket I've created for this.

**Review by helenb — APPROVED** [review]
All working well and LGTM!

File: tbx/project_styleguide/templates/patterns/molecules/pagination/pagination.html [hunk 2]
```diff
@@ -1,38 +1,63 @@
...
-            </li>
-            <li class="pagination__page">
-                <a href="?page=next" class="pagination__link pagination__link--next" rel="next" aria-label="Go to next page">
-                    Next
-                    {% include "patterns/atoms/icons/icon.html" with name="arrow" classname="pagination__next-icon" %}
-                </a>
-            </li>
+            {% if pages.has_previous %}
+                <li class="pagination__page">
+                    <a href="?page={{ pages.previous_page_number }}{% if extra_url_params %}&amp;{{ extra_url_params }}{% endif %}" class="pagination__link pagination__link--previous" rel="prev" aria-label="Go to previous page">
```
Comment (ababic) [inline tbx/project_styleguide/templates/patterns/molecules/pagination/pagination.html:6]: https://pypi.org/project/django-querystring-tag/ is your friend here... no need to worry about setting or managing `extra_url_params`

File: tbx/project_styleguide/templates/patterns/molecules/pagination/pagination.html [hunk 3]
```diff
@@ -1,38 +1,63 @@
...
-                    7
-                </a>
-            </li>
-            <li class="pagination__page">
-                <a href="?page=next" class="pagination__link pagination__link--next" rel="next" aria-label="Go to next page">
-                    Next
-                    {% include "patterns/atoms/icons/icon.html" with name="arrow" classname="pagination__next-icon" %}
-                </a>
-            </li>
+            {% if pages.has_previous %}
```
Comment (ababic) [inline tbx/project_styleguide/templates/patterns/molecules/pagination/pagination.html:4]: The `pages` variable name here is a little confusing. At this point in the code, `pages` is an instance of `django.core.pagination.Page`... `page_obj` or `paginator_page` would be more appropriate.

File: tbx/project_styleguide/templates/patterns/molecules/pagination/pagination.html [hunk 4]
```diff
@@ -1,38 +1,63 @@
 {% if pages.paginator.num_pages > 1 %}
```
Comment (ababic) [inline tbx/project_styleguide/templates/patterns/molecules/pagination/pagination.html:1]: Let's try and stick with established naming conventions for things like pagination. The [examples in the Django docs](https://docs.djangoproject.com/en/5.0/topics/pagination/) have used `page_obj` for the current page for years, which tends to work well. 

If you don't want the 'iterable of results' to be called `page_obj` in the main template, then do what Django's generic views do and make the same object available as both `pages` and `page_obj`.

**Pull request merged**